innkeeper/clientauth.go

```diff
@@ -0,0 +1,32 @@
+package innkeeper
+
+import (
+	"github.com/zalando/skipper/oauth"
+)
+
+// An Authentication object provides authentication to Innkeeper.
+type Authentication interface {
```

Comment:
I guess we can think about it, and fix it in another pull request.

My feeling is that removing abstractions, makes code more monolithic and starting to break the SRP and DIP.